src/interface/lineplot.jl

```diff
@@ -96,13 +97,14 @@ function lineplot(
         y::AbstractVector;
         canvas::Type = BrailleCanvas,
         color::UserColorType = :auto,
-        name = "",
+        name::AbstractString = "",
```

Comment:
@t-bltg Restricting `name` to `AbstractString`, here and below, broke the tests of MCMCChains: https://github.com/TuringLang/AbstractMCMC.jl/runs/4889635592?check_suite_focus=true#step:6:677 This seems like a breaking change?

Reply:
Sorry to heat, this is unfortunate. I don't really see how we should support `Vector{String}` here: for a single series, there should be a single name, no ?

Reply:
Following example works correctly, so I'm guessing invalid data is fed into `Plots` from you recipes in https://github.com/TuringLang/MCMCChains.jl/blob/master/src/plot.jl:

```julia
julia> using Plots; unicodeplots()
julia> x = 1:10; y = rand(10, 2)
julia> plot(x, y, title = "Two Lines", label = ["Line 1" "Line 2"], lw = 3)
               ⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀Two Lines⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀       
               ┌────────────────────────────────────────┐       
         0.992 │⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⣰⠀⠀⠀⠀⠀│ Line 1
               │⠀⢆⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢠⠛⡄⠀⠀⠀⠀│ Line 2
               │⠀⠈⢢⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⡜⠀⡇⠀⠀⠀⠀│       
               │⠀⠀⠀⠣⡀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢰⠁⠀⢸⠀⠀⠀⠀│       
               │⠀⠀⠀⠀⠱⡀⠀⠀⠀⠀⠀⠀⠀⡰⠢⡀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⡎⠀⠀⠘⣄⠔⠊⠀│       
               │⠀⠀⠀⠀⠀⢱⠀⠀⠀⠀⠀⠀⢀⠇⠀⠈⠢⡀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢸⠀⠀⢠⠒⡇⠀⠀⠀│       
               │⠀⠀⠀⠀⠀⠈⡆⠀⠀⠀⠀⠀⢸⠀⠀⠀⠀⠈⡆⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢀⠇⠀⠀⡇⠀⢱⠀⠀⠀│       
               │⠀⠀⠀⠀⠀⠀⢣⠀⠀⠀⠀⠀⡇⠀⠀⠀⠀⠀⢱⠀⠀⠀⠀⠀⠀⢀⡄⠀⢀⣀⠜⠀⠀⢰⠁⠀⠸⡀⠀⠀│       
               │⠀⠤⡀⠀⠀⠀⠘⡄⠀⠀⠀⢰⠁⠀⠀⠀⠀⠀⠀⢇⠀⠀⠀⠀⢀⢎⠼⡊⠁⠀⠀⠀⠀⡎⠀⠀⠀⡇⠀⠀│       
               │⠀⠀⠈⠑⠢⢄⠀⢣⠀⠀⠀⡎⠀⠀⠀⠀⠀⠀⠀⠸⡀⠀⠀⢀⠎⡎⠀⢇⠀⠀⠀⠀⢰⠁⠀⠀⠀⢣⠀⠀│       
               │⠀⠀⠀⠀⠀⠈⢆⠘⡄⠀⢠⠃⠀⠀⠀⠀⠀⠀⠀⠀⢣⠀⢀⠎⡜⠀⠀⠘⡄⠀⠀⠀⡎⠀⠀⠀⠀⢸⠀⠀│       
               │⠀⠀⠀⠀⠀⠀⠈⢆⢇⠀⡜⠀⠀⠀⠀⠀⠀⠀⠀⠀⠈⡦⠎⡜⠀⠀⠀⠀⢱⠀⠀⢠⠃⠀⠀⠀⠀⠀⡇⠀│       
               │⠀⠀⠀⠀⠀⠀⠀⠈⢾⣀⠇⠀⠀⢀⡀⠀⠀⠀⠀⡠⠊⢱⡸⠀⠀⠀⠀⠀⠈⡆⠀⡜⠀⠀⠀⠀⠀⠀⢇⠀│       
               │⠀⠀⠀⠀⠀⠀⠀⠀⠈⢿⠤⠒⠉⠁⠈⠉⠑⠒⠉⠀⠀⠀⠁⠀⠀⠀⠀⠀⠀⠸⣠⠃⠀⠀⠀⠀⠀⠀⠸⠀│       
   -0.00556734 │⣀⣀⣀⣀⣀⣀⣀⣀⣀⣀⣀⣀⣀⣀⣀⣀⣀⣀⣀⣀⣀⣀⣀⣀⣀⣀⣀⣀⣀⣀⣟⣀⣀⣀⣀⣀⣀⣀⣀⣀│       
               └────────────────────────────────────────┘       
               ⠀0.73⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀10.27⠀    
```

Reply:
I'm sorry for the noise, I guess I should have checked the (quite old and probably suboptimal) code in MCMCChains: https://github.com/TuringLang/MCMCChains.jl/pull/353 It took a while to figure out why and which change caused the test errors - and since it "worked" before and still "works" in other Plots backends (e.g., GR just prints "[Line 1, Line 2]" if a vector is provided) I assumed the problem is caused by UnicodePlots or its integration in Plots. Sorry again!

Reply:
No worries, I'm glad to hear this has been fixed on your side and that it's not a regression :)